### PR TITLE
feat(chat): call stage-header (title/duration/connection) + More overflow (#1020)

### PR DESCRIPTION
**Pull request by randax**
## Summary

Implements #1020 (Call Zoom-parity epic #1017): a status-only **stage-header**
for the in-call surfaces and a control-bar restructure that moves less-used
actions under a **More ▸** overflow. Built test-first (red→green per behavior).

## What changed

- **`CallStageHeader.vue`** (new) — status-only header mounted above the stage
  on both the split and expanded surfaces. Shows:
  - the call title,
  - a live elapsed-time timer (counts up once the LiveKit room connects; reads
    "Connecting…" until then),
  - the contextual subline ("Waiting for others…", "N others in call", "Video
    call"),
  - the connection-quality indicator, **relocated here** from the control bar,
  - an inert **recording-indicator slot** — reserved as a present-but-empty
    polite live region (driven `false` until recording lands in a later slice).
- **`call-duration.ts`** (new) — pure `formatCallDuration(ms)` (`M:SS` /
  `H:MM:SS`, clamped/floored) plus the `$callActiveSince` atom and
  `callElapsedMs` helper.
- **`use-call-elapsed.ts`** (new) — composable that ticks once per second and
  formats the elapsed time; browser-only interval with `onUnmounted` cleanup,
  SSR-safe. The start instant lives in the store, so the timer survives the
  split↔expanded remount instead of resetting.
- **`CallControls.vue`** — connection indicator removed (now in the header); the
  Settings action folds into an accessible **More ▸** overflow menu (WAI-ARIA
  menu-button: `aria-haspopup`/`aria-expanded`, `role=menu`/`menuitem`, arrow /
  Home / End / Escape / Tab, outside-click + unmount cleanup). The single
  "Call settings" item opens the existing dialog, which already houses the
  device pickers and the diagnostics/stats section.
- **`use-call-engine.ts`** — `setCallActiveSince(Date.now())` on the engine
  `connected` event, `resetCallActiveSince()` on `disconnected`, mirroring the
  connection-quality lifecycle. `connected` fires once per call (a LiveKit
  reconnect uses the separate `connectionPhaseChanged` signal), so the timer is
  not reset mid-call.
- Old per-surface header markup + styles removed from `CallSplitContainer.vue`
  and `CallExpandedSurface.vue` (no orphaned CSS/imports).

## Acceptance criteria

- [x] Stage-header shows call title, a live elapsed-time timer, and connection quality.
- [x] Connection indicator removed from the control bar (now in the header).
- [x] Recording-indicator slot exists in the header, inert until recording lands.
- [x] Settings moves under a More ▸ overflow; the bar stays uncluttered.
- [x] Timer and header state unit-tested via `bun test`.

## Tests

- `call-duration.test.ts` — formatter edge cases (zero, minute/hour rollover,
  negative clamp, sub-second floor), `$callActiveSince` lifecycle, `callElapsedMs`.
- `call-stage-header.test.ts` — title, subline, Connecting↔timer states,
  relocated connection indicator, inert↔wired recording slot.
- `call-stage-header-surface.test.ts` — the header mounts on a live surface and
  the indicator renders there.
- `call-elapsed-engine.test.ts` — engine `connected`/`disconnected` stamp/clear
  the call clock through the real emitter.
- `call-controls.test.ts` — connection indicator gone from the bar; More ▸
  trigger + `role=menu`/`menuitem` Settings item.

`bun test` (affected files) green; `knip` clean; `astro check` reports no new
errors from these files.

## Decisions (confirmed with reviewer)

- More ▸ exposes a single "Settings" item — the dialog already contains devices
  + diagnostics; the menu is list-based so future items slot in.
- The header keeps the contextual subline next to the timer.

## XEP conformance

Status-only UI change with **no XMPP wire shapes or namespaces** introduced. The
recording slot is inert (no Jingle/recording semantics yet), so there is no XEP
conformance surface in this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: chat/src/components/calls/CallStageHeader.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        {{ props.subline }}
+      </div>
+    </div>
+    <span
+      class="call-stage-header__timer type-control"
+      role="timer"
+      aria-label="Call duration"
+      aria-live="off"
+      aria-atomic="true"
+    >{{ running ? label : "Connecting…" }}</span>
```
[the remark this answers is not in the document]
Comment (randax) [reply chat/src/components/calls/CallStageHeader.vue:62]: Fixed in 7769573d — \`role="timer"\` and \`aria-label\` are now applied only when the timer is running (\`:role="running ? 'timer' : undefined"\`); while connecting it renders as plain status text ("Connecting…"). Pinned with a test.

File: chat/src/lib/calls/call-duration.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { atom } from "nanostores";
+
+/**
+ * Wall-clock instant (ms epoch) at which the active call's LiveKit room
+ * connected, or `null` while no call is connected. The engine stamps it
+ * on `connected` and clears it on `disconnect`, mirroring the
+ * connection-quality lifecycle. The stage-header reads it to drive a
+ * live elapsed-time timer that survives split↔expanded remounts (the
+ * timestamp lives in the store, not in component state).
+ */
```
[the remark this answers is not in the document]
Comment (randax) [reply chat/src/lib/calls/call-duration.ts:10]: Fixed in 7769573d — reworded the doc comment to reference the actual \`connected\`/\`disconnected\` engine events.

File: chat/src/lib/calls/use-call-elapsed.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  callElapsedMs,
+  formatCallDuration,
+} from "./call-duration";
+
+/**
+ * Live elapsed-time clock for the call stage-header.
+ *
+ * Reads the store-owned `$callActiveSince` stamp (set by the engine on
+ * `connected`, cleared on `disconnect`) and re-renders a `M:SS` /
+ * `H:MM:SS` label once per second. Keeping the start instant in the
```
[the remark this answers is not in the document]
Comment (randax) [reply chat/src/lib/calls/use-call-elapsed.ts:14]: Fixed in 7769573d — comment now references the \`disconnected\` event name.

File: chat/src/components/calls/CallControls.vue [hunk 4]
```diff
@@ -214,14 +315,39 @@ function onViewKeydown(event: KeyboardEvent): void {
...
+      >
+        <MoreHorizontal class="w-4 h-4" />
+      </button>
+      <div
+        v-show="moreOpen"
+        ref="moreMenuEl"
+        class="call-controls__more-menu"
+        role="menu"
+        aria-label="More options"
+        @keydown="onMoreMenuKeydown"
```
[the remark this answers is not in the document]
Comment (randax) [reply chat/src/components/calls/CallControls.vue:338]: Good catch — real bug, fixed in 7769573d. \`CallExpandedSurface\`'s capture-phase Escape handler now defers when an overflow menu is open (it checks for an open \`[aria-haspopup="menu"][aria-expanded="true"]\` inside \`.call-controls\`), so Escape closes the More menu via its own bubble-phase handler instead of collapsing the call to split.

File: chat/src/components/calls/CallStageHeader.vue [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    <CallConnectionIndicator />
+    <!-- Inert capture-state slot: a present-but-empty polite live region so
+         that, once capture lands and flips `recording` on, the inserted label
+         is announced. Empty (no dot, no label) until then. -->
+    <span class="call-stage-header__recording" role="status" aria-live="polite">
+      <template v-if="props.recording">
+        <Circle class="call-stage-header__recording-dot" aria-hidden="true" />
+        <span class="type-control">Recording</span>
+      </template>
+    </span>
```
[the remark this answers is not in the document]
Comment (randax) [reply chat/src/components/calls/CallStageHeader.vue:72]: Fixed in 7769573d — the recording dot is now a CSS-filled circle (\`border-radius:9999px; background:currentColor\`) instead of a Lucide \`Circle\`, so it's unambiguously solid with no SVG paint inheritance. Dropped the now-unused \`Circle\` import.

File: chat/src/components/calls/CallStageHeader.vue [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+const { running, label } = useCallElapsed();
+</script>
+
+<template>
+  <header
+    class="call-stage-header"
+    :class="{ 'call-stage-header--compact': props.compact }"
+  >
+    <span class="call-stage-header__live-dot" aria-hidden="true" />
```
[the remark this answers is not in the document]
Comment (randax) [reply chat/src/components/calls/CallStageHeader.vue]: Fixed in 7769573d — the live dot now stays muted + static while connecting and only goes green + pulses once the call clock is running (\`--live\` modifier gated on \`running\`). Pinned with a test.
